base/doc/clsguide.tex

```diff
@@ -977,6 +980,47 @@ \subsection{Passing options around}
 handler, but this handler is not used for |landscape| because that option is
 explicitly declared.
 
+\subsection{Useful status tests}
+
+A number of utility status tests are available which can be used by package and
+class authors to query the presence and date of other code, the current kernel
+release and other related ideas. Three forms of each test are provided, one
+requiring arguments for both \meta{true} and \meta{false} branches, one
+requiring only a \meta{true} branch and one requiring only a \meta{false}
+branch. These are indicated in their names as |TF|, |T| and |F|, respectively.
+Here, we document only the |TF| versions: the other forms are however
+available.
+
+\begin{decl}
+  |\IfPackageLoadedTF| \arg{package-name} \arg{true code} \arg{false code}\\
+  |\IfClassLoadedTF| \arg{class-name} \arg{true code} \arg{false code}\\
+  |\IfFileLoadedTF| \arg{file-name} \arg{true code} \arg{false code}
+\end{decl}
+These commands test if the named package, class or file is loaded: this
+requires that a matching |\Provides...| line has been given, as described in
+Section~\ref{sec:ID}. In the case of \cs{IfFileLoadedTF}, the full
+\meta{file-name} must be provided; in contrast, no extension should be given if
+testing for a package or class.
+
+\begin{decl}
+  |\IfPackageAtLeastTF| \arg{package-name} \arg{date} \arg{true code} \arg{false code}\\
+  |\IfClassAtLeastTF| \arg{class-name} \arg{date} \arg{true code} \arg{false code}\\
+  |\IfFileAtLeastTF| \arg{file-name} \arg{date} \arg{true code} \arg{false code}
+\end{decl}
+These tests are used to check if a package or class has been loaded with at
+least a given \meta{date}. The \meta{date} is compared with that in the
+optional argument to \cs{Provides...} (as described in Section~\ref{sec:ID}).
+As for \cs{Provides...}, the \meta{date} should be given in the form
+`\textsc{yyyy-mm-dd}' or `\textsc{yyyy/nmm/dd}'. If the package or class is not
+loaded, the \meta{false} branch will be taken.
+
+\begin{decl}
+  |\IfFormatAtLeastTF| \arg{date} \arg{true code} \arg{false code}
+\end{decl}
+Tests the release \meta{date} of the \LaTeX{} format and selects the
+appropriate branch. This will reflect any roll forward or back that has been
```

Comment:
Ok, but how will it "reflect" this? 
To what exactly is any rollback applied? and what effect does this have on the date used in the comparison?

Reply:
isn't that what the rest of the sentence says? e.g if you roll back to yyyy/mm/dd then the conditional uses that date as the format date for comparison.

Reply:
Yes, the explanation is indeed there.

But the first part needs to be more precise, e.g. thus:

The date used for the format is adjusted to incorporate any rollback that has been applied to it, . . .